packages/bff/src/utilities/fileutils.ts

```diff
@@ -0,0 +1,210 @@
+/* eslint-disable functional/immutable-data */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable max-params */
+import path from "path";
+import AdmZip from "adm-zip";
+import { catalogApi } from "pagopa-interop-api-clients";
+import { FileManager, Logger } from "pagopa-interop-commons";
+import { genericError } from "pagopa-interop-models";
+import { missingInterface } from "../model/domain/errors.js";
+import { retrieveEserviceDescriptor } from "../model/modelMappingUtils.js";
+import { verifyExportEligibility } from "../model/validators.js";
+/* 
+  FileDocumentsRegistry is a map that contains the following information:
+  - occurrences: a map that contains the number of occurrences of a document name
+    (the same document name can be used multiple times in the same descriptor)
+  - uniqueNames: a map that contains the unique name for each document id
+*/
+export type FileDocumentsRegistry = {
+  occurrences: Map<string, number>;
+  uniqueNames: Map<string, string>;
+};
+
+export type FileData = {
+  id: string;
+  file: Uint8Array;
+};
+
+function getUniqueNameByDocumentId(
+  fileDocumentRegistry: FileDocumentsRegistry,
+  documentId: string
+): string {
+  const uniqueName = fileDocumentRegistry.uniqueNames.get(documentId);
+  if (!uniqueName) {
+    throw genericError(`Unique name not found for document id ${documentId}`);
+  }
+  return uniqueName;
+}
+
+/* 
+  This function creates a FileDocumentsRegistry object that contains the following information:
+  - occurrences: a map that contains the number of occurrences of a document name
+    (the same document name can be used multiple times in the same descriptor)
+  - uniqueNames: a map that contains the unique name for each document id
+*/
+export function buildFileDocumentRegistry(
+  eserviceDocuments: catalogApi.EServiceDoc[]
+): FileDocumentsRegistry {
+  return eserviceDocuments.reduce(
+    (fileRegistry: FileDocumentsRegistry, doc: catalogApi.EServiceDoc) => {
+      const originalName = doc.name;
+      const occurrence = fileRegistry.occurrences.get(doc.name) || 0;
+      fileRegistry.occurrences.set(originalName, occurrence + 1);
+
+      const newName = occurrence
+        ? `${path.parse(originalName).name}-${occurrence}${path.extname(
+            originalName
+          )}`
```

Comment:
```suggestion
      const parsedName = path.parse(originalName);

      const newName = occurrence
        ? `${parsedName.name}-${occurrence}${parsedName.ext}`
```

Reply:
solved in this [commit](https://github.com/pagopa/interop-be-monorepo/pull/882/commits/29160dd1e958ed8997f0e89f1b2cef3f8e959fc6)